File: JavaScript Advanced/Arrays and Nested Arrays/ticTacToe.js
function solve(moves) {
    function isWinner(sign, matrixField) {
        // First row
        if (matrixField[0][0] === sign && matrixField[0][1] === sign && matrixField[0][2] === sign) return true;
        // Second row
        else if (matrixField[1][0] === sign && matrixField[1][1] === sign && matrixField[1][2] === sign) return true;
        // Third row
        else if (matrixField[2][0] === sign && matrixField[2][1] === sign && matrixField[2][2] === sign) return true;
        // First column
        else if (matrixField[0][0] === sign && matrixField[1][0] === sign && matrixField[2][0] === sign) return true;
        // Second column
        else if (matrixField[0][1] === sign && matrixField[1][1] === sign && matrixField[2][1] === sign) return true;
        // Third column
        else if (matrixField[0][2] === sign && matrixField[1][2] === sign && matrixField[2][2] === sign) return true;
        // First diagonal
        else if (matrixField[0][0] === sign && matrixField[1][1] === sign && matrixField[2][2] === sign) return true;
        // Second diagonal
        else if (matrixField[0][2] === sign && matrixField[1][1] === sign && matrixField[2][0] === sign) return true; else return false;
    }

    let field = [[false, false, false], [false, false, false], [false, false, false]]

    let playerCounter = 1;
    let freeSpace = 9;
    let playerSign;

    for (let move of moves) {
        let [moveRow, moveCol] = move.split(' ').map(x => Number(x));
        playerCounter % 2 === 1 ? playerSign = 'X' : playerSign = 'O';

        if (field[moveRow][moveCol] !== false) {
            console.log('This place is already taken. Please choose another!');
            continue;
        }

        field[moveRow][moveCol] = playerSign;
        playerCounter++;
        freeSpace--;
        if (isWinner(playerSign, field)) {
            console.log(`Player ${playerSign} wins!`);
            break;
        }

        if (freeSpace === 0) {
            console.log('The game ended! Nobody wins :(');
            break;
        }
    }
    for (let row of field) console.log(row.join('\t'));
}

File: Python Fundamentals/Lists/tic_tac_toe.py
all_lines = [input().split(), input().split(), input().split()]

all_rows = [
    [all_lines[0][0], all_lines[1][0], all_lines[2][0]],
    [all_lines[0][1], all_lines[1][1], all_lines[2][1]],
    [all_lines[0][2], all_lines[1][2], all_lines[2][2]]
]

all_diagonals = [
    [all_lines[0][0], all_lines[1][1], all_lines[2][2]],
    [all_lines[0][2], all_lines[1][1], all_lines[2][0]]
]

for line in all_lines:
    if line.count('1') == len(line):
        print("First player won")
        exit()
    elif line.count('2') == len(line):
        print("Second player won")
        exit()

for row in all_rows:
    if row.count('1') == len(row):
        print("First player won")
        exit()
    elif row.count('2') == len(row):
        print("Second player won")
        exit()

for diagonal in all_diagonals:
    if diagonal.count('1') == len(diagonal):
        print("First player won")
        exit()
    elif diagonal.count('2') == len(diagonal):
        print("Second player won")
        exit()

print('Draw!')
